src/options.rs

```diff
@@ -107,3 +140,70 @@ impl Default for RedirectPolicy {
         RedirectPolicy::None
     }
 }
+
+/// Possible certificate archive file formats.
+///
+/// If a format is not supported by the underlying SSL/TLS engine, an error will
+/// be returned when attempting to send a request using the offending
+/// certificate.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum CertificateFormat {
+    /// A DER-encoded certificate file.
+    DER,
+    /// A PEM-encoded certificate file.
+    PEM,
+    /// A PKCS#12-encoded certificate file.
+    P12,
+}
+
+impl Default for CertificateFormat {
+    fn default() -> Self {
+        CertificateFormat::PEM
+    }
+}
+
+impl std::fmt::Display for CertificateFormat {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            CertificateFormat::DER => f.write_str("DER"),
+            CertificateFormat::PEM => f.write_str("PEM"),
+            CertificateFormat::P12 => f.write_str("P12"),
+        }
+    }
+}
+
+/// A public key certificate file.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Certificate {
+    pub(crate) path: PathBuf,
+    pub(crate) format: CertificateFormat,
+}
+
+impl Certificate {
+    /// Create a new certificate object from the given path.
+    pub fn new(path: impl Into<PathBuf>, format: CertificateFormat) -> Self {
+        Self {
+            path: path.into(),
+            format,
+        }
+    }
+}
+
+/// A private key file.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct PrivateKey {
+    pub(crate) path: PathBuf,
+    pub(crate) format: CertificateFormat,
+    pub(crate) password: String,
```

Comment:
Not every private key file has a password. I think most PKCS12-encoded key pairs will, but at least with PEM formats its not unusual to have an unencrypted key.